### ISSUE TITLE
Shortcuts with R functions

### ISSUE THREAD

**Issue by MaTo04** [Ikuyadeu/vscode-R#100]
First of all, thanks for this VS Code extension. I came from Tinn-R, the big advantage from this is to define  shortcuts for R-code like nrow(selection) or length(selection). I know how to run Selection by "ctrl + Enter". But is it currently possible to define Shortcuts with 'R-Code' and automatically send it to the console ? 

I think it´s possible to adapt or copy the function 'runSelection' from the extension.ts. Therefore two thinks have to be changed in my opinion. First of all runSelection needs a input variable (simple function name like 'nrow'). Secondly the `rTerm.sendText(line)` has to be change:

`async function runSelection(RfunctionName) {`
`var first = true`
`for (const line of selection.selectedTextArray) {`
` if (checkForComment(line)) { continue; }`
` await delay(8); // Increase delay if RTerm can't handle speed.`
` if (first) {rTerm.sendText(RfunctionName + '('}`
` if (last) {')'}`
` rTerm.sendText(line);`
`}`

I`m not familiar with Typescript but i think this could be the way to define R-Shortcuts within this Extension.

**Comment by MaTo04** [comment]
Due to a missing reaction i adapt the 'extension.js' and 'package.json' to use R-Commands in Visual Studio Code. Now i would like to explain my solution for the example 'nrow'. First of all in the package.json the "commands" and "keybindings" has to be changed. Sencondly the function 'runSelection' and 'context.subscriptions' has to be changed:

### package.json
**commands:** 
`{"title": "Show number of rows for selected object", "category": "R", "command": "r.nrow"}`

**keybindings:**
`{"command": "r.nrow", "key": "Ctrl+1","mac": "cmd+1","when": "editorTextFocus && editorLangId == 'r'"},`

### extension.js
**runSelection**
```
async function runSelection(rFunctionName) {
    const selection = selection_1.getSelection();
    if (!rTerminal_1.rTerm) {
        const success = rTerminal_1.createRTerm(true);
        if (!success) {
            return;
        }
        await util_1.delay(200); // Let RTerm warm up
    }
    if (selection.linesDownToMoveCursor > 0) {
        vscode_1.commands.executeCommand("cursorMove", { to: "down", value: selection.linesDownToMoveCursor });
        vscode_1.commands.executeCommand("cursorMove", { to: "wrappedLineEnd" });
    }

    for (let line of selection.selectedTextArray) {
        if (selection_1.checkForComment(line)) {
            continue;
        }
        await util_1.delay(8); // Increase delay if RTerm can't handle speed.

        if (rFunctionName) {
            let rFunctionCall = ""
            for (const feature of rFunctionName) {
                rFunctionCall += feature + "("  
            }
            line = rFunctionCall + line.trim() + ")".repeat(rFunctionName.length);
        }
        rTerminal_1.rTerm.sendText(line);
    }
    setFocus(); 	
}
```




**context.subscriptions**
`context.subscriptions.push(vscode_1.commands.registerCommand("r.nrow", () => runSelection(["nrow"])))`

If somebody is interersted in this solution feel free to use it. If you are interested i would send you the adapted files with some new keybindings like: nrow, length, head, transposed head, names. runSelection accepts empty input (standard runSelection like before), single input or an array of R-functions like transposed head = t(head(*selected*)). Every line give separated output. If two variables in two lines are selected, the function give for every line the nrow or something else.
This adaption now allows us to use VS Code like Tinn-R with shortcuts.

**Comment by Ikuyadeu** [comment]
@MaTo04 Thank you for your suggestion, It's a great job!
If you will submit pull request I'll merge this (I'll put your name on changelog).
But if you would not, I want to just copy your code.
Which do you want to do?

**Comment by MaTo04** [comment]
Thanks, i will submit pull request today.

**Comment by Ikuyadeu** [comment]
Fixed by #101, thank you @MaTo04
